neo/Network/LocalNode.cs

```diff
@@ -277,38 +278,77 @@ private static void CheckMemPool()
 
         private void ConnectToPeersLoop()
         {
+            List<Task> tasksList = new List<Task>();
+            DateTime lastSufficientPeersTimestamp = DateTime.UtcNow;
+
             while (!cancellationTokenSource.IsCancellationRequested)
             {
                 int connectedCount = connectedPeers.Count;
                 int unconnectedCount = unconnectedPeers.Count;
                 if (connectedCount < ConnectedMax)
                 {
-                    Task[] tasks = { };
                     if (unconnectedCount > 0)
                     {
                         IPEndPoint[] endpoints;
                         lock (unconnectedPeers)
                         {
                             endpoints = unconnectedPeers.Take(ConnectedMax - connectedCount).ToArray();
                         }
-                        tasks = endpoints.Select(p => ConnectToPeerAsync(p)).ToArray();
+                        tasksList.AddRange(endpoints.Select(p => ConnectToPeerAsync(p)));
                     }
-                    else if (connectedCount > 0)
+
+                    if (connectedCount > 0)
                     {
-                        lock (connectedPeers)
+                        if (unconnectedCount + connectedCount < DesiredAvailablePeers)
+                        {
+                            lock (connectedPeers)
+                            {
+                                foreach (RemoteNode node in connectedPeers)
+                                    node.RequestPeers();
+                            }
+
+                            if (lastSufficientPeersTimestamp < DateTime.UtcNow.AddSeconds(-180))
+                            {
+                                Random rand = new Random();
+                                tasksList.AddRange(Settings.Default.SeedList
+                                    .OrderBy(p => rand.Next()).Take(2)
+                                    .OfType<string>().Select(p => p.Split(':'))
+                                    .Select(p => ConnectToPeerAsync(p[0], int.Parse(p[1]))));
```

Comment:
I could filter out the currently connected nodes from this list so there is no potential that it could try to connect to an already connected node; however, the code that handles connecting will take care of it in that case, so seems fine as is.